src/v/cluster/cluster_discovery.cc

```diff
@@ -281,6 +292,53 @@ std::optional<int> get_node_index_in_seed_servers() {
     return std::nullopt;
 }
 
+void verify_bootstrap_info_consistency(
+  bool& failed,
+  const cluster::cluster_bootstrap_info_reply& reply,
+  const net::unresolved_address& seed_addr) {
+    vassert(!reply.cluster_uuid.has_value(), "Should have returned early");
+    if (
+      reply.version != features::feature_table::get_latest_logical_version()) {
+        vlog(
+          clusterlog.error,
+          "logical version mismatch, local: {}, {}: {}",
+          features::feature_table::get_latest_logical_version(),
+          seed_addr,
+          reply.version);
+        failed = true;
+    }
+    if (!equal(reply.seed_servers, config::node().seed_servers())) {
+        vlog(
+          clusterlog.error,
+          "seed server list mismatch, local: {}, {}: {}",
+          config::node().seed_servers(),
+          seed_addr,
+          reply.seed_servers);
+        failed = true;
+    }
+    if (
+      reply.empty_seed_starts_cluster
+      != config::node().empty_seed_starts_cluster()) {
+        vlog(
+          clusterlog.error,
+          "empty_seed_starts_cluster mismatch, local: {}, {}: {}",
+          config::node().empty_seed_starts_cluster(),
+          seed_addr,
+          reply.empty_seed_starts_cluster);
+        failed = true;
+    }
+    if (seed_addr != reply.broker.rpc_address()) {
+        vlog(
+          clusterlog.warn,
+          "advertised_rpc_api mismatch, local (as per seed_servers): {}, "
+          "remote (as per advertised_rpc_api): {}. This is not an error and "
+          "the cluster will proceed, however the cluster will have less actual "
+          "seed nodes that expected, and may even fail to bootstrap",
```

Comment:
I think those are irrelevant to this PR. There will be less seed nodes because the one with different `avertised_rpc_api` will never act as a seed node (`get_node_index_in_seed_servers()` will always return empty). The cluster will fail to bootstrap if the number of misconfigured nodes is at least a half of the total.

Reply:
> will never act as a seed node

Ah good point about a misconfigured majority.